### PR TITLE
Plugin architecture for the transform panel, add babel live coding support

### PR DESCRIPTION
**Pull request by cpojer**
See #26 

@gaearon requested it here: https://twitter.com/dan_abramov/status/638659321228840960

Should have built it like this from the beginning.

There are some minor annoyances with toggling transformers – how do we alert the user that toggling means they have to rewrite their code? I made it so it only wipes away the default transform code but the code there is slightly messy. I'm sure @fkling can make it much much better.

Also probably needs an update so that Parse saves the used transform plugin.

### CODE REVIEW

**Comment by gaearon** [comment]
This. Is. So. Rad! I love it.

I see a dropdown glitch in Chrome:

screen shot 2015-09-01 at 21 46 32

**Comment by cpojer** [comment]
Hey now, don't hate on my CSS.

File: src/transformers/jscodeshift.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+const defaultTransform =
+  fs.readFileSync(__dirname + '/transformJscodeshift.txt', 'utf8');
+
+function transform({transformCode, code}) {
+  return new Promise((resolve, reject) => {
+    loadjs(['babel-core', 'jscodeshift'], (babel, jscodeshift) => {
+      try {
+        // This might throw
+        var transform = compileModule(
+          babel.transform(transformCode).code
```
Comment (cpojer) [inline src/transformers/jscodeshift.js]: I felt really tempted to self-host the plugin with a previous version of the transformCode, like

```
babel.transform(transformCode, {plugins: [previousTransformVersion]});
```

:lollipop:

@sebmck

**Comment by fkling** [comment]
I guess this is a good first strep. My actual idea was to abstract even more so that "tools" can specify their own view. E.g. for eslint stuff it will be necessary to have a panel to define options.

However simple transform tools could build on this. And yeah, I have to update the parse code as well.

**Comment by fkling** [comment]
Can you just not commit the `dist/` folder? I will rebuild anyway.

File: src/transformers/utils/compileModule.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+export default function compileModule(code) {
+  var m = {};
+  var f = new Function('module', `(function() {var exports = {}; ${code}}())`);
```
Comment (DmitrySoshnikov) [inline src/transformers/utils/compileModule.js]: Can you assign to the local `exports`? How is it returned?

Comment (cpojer) [inline src/transformers/utils/compileModule.js]: `module.exports = exports;` is added by babel.

**Comment by cpojer** [comment]
@fkling got rid of the dist folder from the commit. Am I not supposed to commit it like I used to?

**Comment by fkling** [comment]
I will rebuild anyway.

**Pull request closed without merging**